distributed-testing/src/main/groovy/net/corda/testing/DistributedTesting.groovy

```diff
@@ -0,0 +1,308 @@
+package net.corda.testing
+
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+import com.bmuschko.gradle.docker.tasks.image.DockerPushImage
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.testing.Test
+import org.gradle.api.tasks.testing.TestResult
+
+import java.util.stream.Collectors
+
+/**
+ This plugin is responsible for wiring together the various components of test task modification
+ */
+class DistributedTesting implements Plugin<Project> {
+
+    public static final String GRADLE_GROUP = "Distributed Testing";
+
+    static def getPropertyAsInt(Project proj, String property, Integer defaultValue) {
+        return proj.hasProperty(property) ? Integer.parseInt(proj.property(property).toString()) : defaultValue
+    }
+
+    @Override
+    void apply(Project project) {
+        if (System.getProperty("kubenetize") != null) {
```

Comment:
Why use a Java system property when Gradle has its own properties?

Reply:
Grade properties are System properties.

Reply:
Are you sure about that, @roastario ? Try running this:
```gradle
plugins {
    id 'base'
}

if (project.hasProperty('user.home')) {
    println "User Home=${project.property('user.home')}"
}

if (project.hasProperty('my.message')) {
    println "My Messagee=${project.property('my.message')}"
}

[ 'user.home', 'my.message' ].forEach { p ->
    println "Java '$p' System Property: ${System.getProperty(p)}"
}


wrapper {
    gradleVersion = "5.4.1"
    distributionType = Wrapper.DistributionType.ALL
}
```
I get the following output:
```bash
$ ./gradlew -Pmy.message='Hello Stefano' clean

> Configure project :
My Messagee=Hello Stefano
Java 'user.home' System Property: /home/chris
Java 'my.message' System Property: null

BUILD SUCCESSFUL in 0s
1 actionable task: 1 up-to-date
```